packages/ui5-library-writer/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import { join } from 'path';
+import { create as createStorage } from 'mem-fs';
+import type { Editor } from 'mem-fs-editor';
+import { create } from 'mem-fs-editor';
+import cloneDeep from 'lodash/cloneDeep';
+import type { UI5LibConfig } from './types';
+import { enableTypescript } from './options';
+import { mergeWithDefaults } from './data';
+
+/**
+ * Writes the template to the memfs editor instance.
+ *
+ * @param basePath - the base path
+ * @param ui5LibConfig - the Ui5 lib instance
+ * @param fs - the memfs editor instance
+ * @returns the updated memfs editor instance
+ */
+async function generate(basePath: string, ui5LibConfig: UI5LibConfig, fs?: Editor): Promise<Editor> {
+    const reuseLib = cloneDeep(ui5LibConfig);
+
+    if (!fs) {
+        fs = create(createStorage());
+    }
+
+    const libInput = mergeWithDefaults(ui5LibConfig);
+    basePath = join(basePath, libInput.libraryNamespace);
+
+    const tmplPath = join(__dirname, '..', 'templates');
+    const ignoreJSFiles = ['**/src/baselibrary/Example.js', '**/ExampleRenderer.js', '**/library.js'];
```

Comment:
Here, I also needed a moment to understand why you list these specific js files and in contrast all ts files. If I got it, it is because the test files are js even in case of ts.
Is there a reason why we don't use ts for testing?

Reply:
The files are converted to ts